include/systems/system.h

```diff
@@ -1596,6 +1597,17 @@ class System : public ReferenceCountedObject<System>,
    * User code can use this for accumulating error estimates for example.
    */
   std::vector<Number> qoi_error_estimates;
+  #endif
+
+  /** Accessors for qoi and qoi_error_estimates vectors
+   */
+  void init_qois(unsigned int n_qois);
+
+  void fill_qoi(unsigned int qoi_index, Real qoi_value);
+  Real get_qoi_value(unsigned int qoi_index);
+
+  void fill_qoi_error_estimate(unsigned int qoi_index, Real qoi_error_estimate);
```

Comment:
This is slightly more interesting.  We want to be able to store directional error estimates here, e.g. for refinement-based estimators that can actually give us more than just a bound?  If so then the API fix is simple, Real->Number again.